code/modules/antagonists/clock_cult/scriptures/structures/stargazer.dm

```diff
@@ -0,0 +1,188 @@
+	summoned_structure = /obj/structure/destructible/clockwork/gear_base/stargazer
+	cogs_required = 2
+	category = SPELLTYPE_STRUCTURES
+
+//Stargazer light
+
+/obj/effect/stargazer_light
+	icon = 'icons/obj/clockwork_objects.dmi'
+	icon_state = "stargazer_closed"
+	pixel_y = 10
+	layer = FLY_LAYER
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	alpha = 160
+	var/active_timer
+
+/obj/effect/stargazer_light/ex_act()
+	return
+
+/obj/effect/stargazer_light/Destroy(force)
+	cancel_timer()
+	. = ..()
+
+/obj/effect/stargazer_light/proc/finish_opening()
+	icon_state = "stargazer_light"
+	active_timer = null
+
+/obj/effect/stargazer_light/proc/finish_closing()
+	icon_state = "stargazer_closed"
+	active_timer = null
+
+/obj/effect/stargazer_light/proc/open()
+	icon_state = "stargazer_opening"
+	cancel_timer()
+	active_timer = addtimer(CALLBACK(src, .proc/finish_opening), 2, TIMER_STOPPABLE | TIMER_UNIQUE)
+
+/obj/effect/stargazer_light/proc/close()
+	icon_state = "stargazer_closing"
+	cancel_timer()
+	active_timer = addtimer(CALLBACK(src, .proc/finish_closing), 2, TIMER_STOPPABLE | TIMER_UNIQUE)
+
+/obj/effect/stargazer_light/proc/cancel_timer()
+	if(active_timer)
+		deltimer(active_timer)
+
+#define STARGAZER_COOLDOWN 1800
+
+//Stargazer structure
+/obj/structure/destructible/clockwork/gear_base/stargazer
+	name = "stargazer"
+	desc = "A small pedestal, glowing with a divine energy."
+	clockwork_desc = "A small pedestal, glowing with a divine energy. Used to provide special powers and abilities to items."
+	default_icon_state = "stargazer"
+	anchored = TRUE
+	break_message = "<span class='warning'>The stargazer collapses.</span>"
+	var/cooldowntime = 0
+	var/mobs_in_range = FALSE
+	var/fading = FALSE
+	var/obj/effect/stargazer_light/sg_light
+
+/obj/structure/destructible/clockwork/gear_base/stargazer/Initialize()
+	. = ..()
+	sg_light = new(get_turf(src))
+	START_PROCESSING(SSobj, src)
+
+/obj/structure/destructible/clockwork/gear_base/stargazer/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	if(!QDELETED(sg_light))
+		qdel(sg_light)
+	. = ..()
+
+/obj/structure/destructible/clockwork/gear_base/stargazer/process()
+	if(QDELETED(sg_light))
+		return
+	var/mob_nearby = FALSE
+	for(var/mob/living/M in view(2, get_turf(src)))
+		if(is_servant_of_ratvar(M))
+			mob_nearby = TRUE
+			break
+	if(mob_nearby && !mobs_in_range)
+		sg_light.open()
+		mobs_in_range = TRUE
+	else if(!mob_nearby && mobs_in_range)
+		mobs_in_range = FALSE
+		sg_light.close()
+
+/obj/structure/destructible/clockwork/gear_base/stargazer/attackby(obj/item/I, mob/living/user, params)
+	if(!is_servant_of_ratvar(user) || user.a_intent != INTENT_HELP)
+		. = ..()
+		return
+	if(!anchored)
+		to_chat(user, "<span class='brass'>You need to anchor [src] to the floor with your dagger first.</span>")
+		return
+	if(cooldowntime > world.time)
+		to_chat(user, "<span class='brass'>The [src] is still warming up, it will be ready in [DisplayTimeText(cooldowntime - world.time)].</span>")
```

Comment:
```suggestion
		to_chat(user, "<span class='brass'>\The [src] is still warming up, it will be ready in [DisplayTimeText(cooldowntime - world.time)].</span>")
```